tools/rpm2archive.c

```diff
@@ -141,10 +163,20 @@ static int process_package(rpmts ts, const char * filename)
 	    exit(EXIT_FAILURE);
 	}
     }
-    if (archive_write_set_format_pax_restricted(a) != ARCHIVE_OK) {
-	fprintf(stderr, "Error: Format pax restricted is not supported\n");
+
+    if (rstreq(format, "pax")) {
+	format_code = ARCHIVE_FORMAT_TAR_PAX_RESTRICTED;
+    } else if (rstreq(format, "cpio")) {
+	format_code = ARCHIVE_FORMAT_CPIO_SVR4_NOCRC;
+    } else {
+	rc = ARCHIVE_FAILED;
+    }
+
+    if (archive_write_set_format(a, format_code) != ARCHIVE_OK) {
+	fprintf(stderr, "Error: Format %s is not supported\n", format);
```

Comment:
There seems to be no `format` variable, isn't this a forgotten "need to get the format name from `format_code`"?

Reply:
... as I suppose using `format_code` wouldn't be optimal either as it's just an integer code.